### PR TITLE
Fix #35

### LINKED ISSUE #35: Overwrite references
Allow users to overwrite references used in the rules. For example, if you have the following

```
mappings:
  person:
    po:
      - [schema:name, $(name)]
```
then you could (either via the library or CLI) say that `name` is always "John".
This means that `name` is no longer taken from the data source. Actually, the rules are then (internally) rewritten to
```
mappings:
  person:
    po:
      - [schema:name, John]
```

@bjdmeest What do you think?

### PR DESCRIPTION
**Pull request by pheyvaer**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,13 +27,15 @@ They are converted to a single RML document.
...
+You can overwrite external references via the `-e`.
+For example, `-e name=John` will replace all occurences of `$(_name)` with `John`.
+Repeat `-e` for multiple references.
 
 #### yarrrml-generator
 
 If you want to generate YARRRML rules from an RML document, you do the following: `yarrrml-generator -i rules.rml.ttl`.
 The rules will be written to standard output.
 If you want to write them to a file, you can add the `-o` option.
-If you want to use `rr:class` instead of Predicate Object Maps, use the `-c` flag.
```
Comment (bjdmeest) [inline README.md:36]: Shouldn't this comment be moved to previous section instead of removed?

File: README.md [hunk 2]
```diff
@@ -27,13 +27,15 @@ They are converted to a single RML document.
 Note that the keys in `prefixes`, `sources`, and `mappings` have to be unique across all files.
 `base` can only be set once.
 An example can be found in [`test/multiple-input-files`](test/multiple-input-files).
+You can overwrite external references via the `-e`.
+For example, `-e name=John` will replace all occurences of `$(_name)` with `John`.
+Repeat `-e` for multiple references.
```
Comment (bjdmeest) [inline README.md:32]: I would add a huge disclaimer that if your reference starts with `_` you should add a `\` before it. This is breaking behavior. Also we need to (separately) make sure the spec is updated accordingly.

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Allow multiple input files for the CLI  (see [issue 34](https://github.com/RMLio/yarrrml-parser/issues/34))
 - reference formulation for CSS  (see [issue 37](https://github.com/RMLio/yarrrml-parser/issues/37))
 - Allow multiple input files for the CLI  (see [issue 34](https://github.com/RMLio/yarrrml-parser/issues/34))
+- Overwrite references  (see [issue 35](https://github.com/RMLio/yarrrml-parser/issues/35))
```
Comment (bjdmeest) [inline CHANGELOG.md:15]: I would add another line under `Changed` that `_id` references now need to be prefixed as `\_id`, so we remember to do the major version update :)

File: lib/abstract-generator.js [hunk 4]
```diff
@@ -673,6 +679,29 @@ class AbstractGenerator {
...
+   */
+  _replaceExternalReferences(str) {
+    str =  '' + str;
+    const refs = str.match(/\{([^\}]*)\}/g);
+
+    if (refs) {
+      for (const ref of refs) {
+        if (ref.startsWith('{_')) {
+          const refWithoutBrackets = ref.substr(2, ref.length - 3);
+          const externalRefValue = this.externalReferences[refWithoutBrackets];
```
Comment (bjdmeest) [inline lib/abstract-generator.js:694]: What if the external reference isn't defined? if not, I'd fall back to keeping it a reference (so basically prefix with `\` here) and log it (and as such, we're not creating a breaking change, which we could deprecate in the future)

File: README.md (hunk 2 above)
Comment (bjdmeest) [reply README.md:32]: See my comment at `lib/abstract-generator.js` to make this a non-breaking feature. I would document as such here then as well

File: README.md (hunk 1 above)
Comment (pheyvaer) [reply README.md:36]: It's already in the yarrrml-parser section. And here it doesn't work that option, right?

File: lib/abstract-generator.js (hunk 4 above)
Comment (pheyvaer) [reply lib/abstract-generator.js:694]: Good point!

File: README.md (hunk 1 above)
Comment (bjdmeest) [reply README.md:36]: ah ok, then it's fine!

**Pull request merged**